### DISCUSSION TITLE
What are the hardware prerequisites to run EGSnrc, what is the best hardware (processor, memory, storage)?

### DISCUSSION THREAD

**Discussion by jcampbell555** [nrc-cnrc/EGSnrc#765] [Questions] [answered]
Out of curiosity, are there specific PC specs that would be ideal for running EGSnrc? RAM, Processor, storage, etc?

**Comment by ftessier** [comment, marked as answer]
An EGSnrc simulation does not require any particular hardware in and of itself, and will likely run on any computer where it can be compiled (it is written in standard Fortran, C and C++).

Some applications such as modelling large voxelized phantom may require a large amount of RAM, while others storing large amounts of particle data in phase space files may require large amounts of disk space. But more often than not, the number of processors threads and their speed is the limiting factor. The Monte Carlo method is pleasingly parallel (each history can be handled separately and asynchronously), hence the performance of a simulation scales ~linearly with the number of CPU cores.

It is difficult to provide overarching guidance other than that, as with other programs in general, a better CPU with a larger _cache_ memory will probably have the most significant impact. Monte Carlo is essentially a brute force floating-point calculation problem , so that is the metric to look for when comparing CPU benchmarks.

Perhaps the best advice, in a way, is to start with a mid-range off-the-shelf computer and test the problem at hand to gauge whether the calculation is CPU-bound, RAM-bound, or I/O-bound, and upgrade accordingly. A common setup in research is to build, test and debug the simulation on a personal computer, and then connect to high performance computing clusters to run final _production_ simulations in decent time.

**Reply by rtownson** [reply]
Some users have found that it cannot compile on Windows 10 systems with 4GB of RAM or less. On linux, you can get away with less.
